hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/CassDefinitionsServiceImpl.java

```diff
@@ -715,6 +938,40 @@ public Trigger getTrigger(String tenantId, String triggerId) throws Exception {
         }
     }
 
+    // This impl is not efficient. It actually pulls all triggers for the tenant and then filters out the
+    // triggers that are not the requested members.  To make this more efficient we'd likely need to
+    // create a new member_triggers table which duplicated the triggers table, and where we would maintain
+    // a replica entry for each member trigger.  We'd need a primary key like ((tenantId,groupId),memberId) and
+    // then we could query for member triggers directly.  Because this method is expected to be called
+    // rarely, and because the trigger population is not expected to be huge (likely maxing out in the thousands),
+    // we may just get away with this dumb impl.
```

Comment:
This method should be used rarely, so I'm not fan to do early improvements until, too.
Also, after reading until here, I'm wondering that as group Triggers are not going to be mixed with normal Triggers, perhaps group triggers can go in a different table.